main/pong.go

```diff
@@ -0,0 +1,587 @@
+
+
+type PongTermMessage struct {
+	Text   string
+	Source string
+}
+
+func initPongSerial() bool {
+	var device string
+	baudrate := int(3000000)
+
+	log.Printf("Configuring Pong ADS-B\n")
+
+	if _, err := os.Stat("/dev/pong"); err == nil {
+		device = "/dev/pong"
+	} else {
+		log.Printf("No suitable Pong device found.\n")
+		return false
+	}
+	log.Printf("Using %s for Pong\n", device)
+
+	// Open port
+	// No timeout specified as Pong does not heartbeat
+	pongSerialConfig = &serial.Config{Name: device, Baud: baudrate}
+	p, err := serial.OpenPort(pongSerialConfig)
+	if err != nil {
+		log.Printf("Error opening serial port: %s\n", err.Error())
+		return false
+	}
+	errRts := p.ClearRTS()
+	if errRts != nil {
+		log.Printf("Clearing RTS returned %s\n", errRts.Error())
+	}
+
+	log.Printf("Pong opened serial port at %d baud\n",baudrate)
+
+	// No device configuration is needed, we should be ready
+	globalStatus.Pong_Heartbeats = 0
+
+	pongSerialPort = p
+	return true
+}
+
+func pongNetworkRepeater() {
+	defer pongWG.Done()
+	log.Println("Entered Pong network repeater ...")
+	cmd := exec.Command(STRATUX_HOME + "/bin/dump1090", "--net-only", "--net-stratux-port", "30006")
+	stdout, _ := cmd.StdoutPipe()
+	stderr, _ := cmd.StderrPipe()
+
+	err := cmd.Start()
+	if err != nil {
+		log.Printf("Error executing "+STRATUX_HOME+"/bin/dump1090: %s\n", err)
+		// don't return immediately, use the proper shutdown procedure
+		shutdownPong = true
+		for {
+			select {
+			case <-closeChpong:
+				return
+			default:
+				time.Sleep(1 * time.Second)
+			}
+		}
+	}
+
+	log.Println("Executed " + cmd.String() + " successfully...")
+
+	scanStdout := bufio.NewScanner(stdout)
+	scanStderr := bufio.NewScanner(stderr)
+
+	for {
+		select {
+		case <-closeChpong:
+			log.Println("Pong network repeater: shutdown msg received, calling cmd.Process.Kill() ...")
+			err := cmd.Process.Kill()
+			if err != nil {
+				log.Printf("\t couldn't kill dump1090: %s\n", err)
+			} else {
+				cmd.Wait()
+				log.Println("\t kill successful...")
+			}
+			return
+		default:
+			for scanStdout.Scan() {
+				m := Dump1090TermMessage{Text: scanStdout.Text(), Source: "stdout"}
+				logDump1090TermMessage(m)
+			}
+			if err := scanStdout.Err(); err != nil {
+				log.Printf("scanStdout error: %s\n", err)
+			}
+
+			for scanStderr.Scan() {
+				m := Dump1090TermMessage{Text: scanStderr.Text(), Source: "stderr"}
+				logDump1090TermMessage(m)
+			}
+			if err := scanStderr.Err(); err != nil {
+				log.Printf("scanStderr error: %s\n", err)
+			}
+			for scanStderr.Scan() {
+				m := Dump1090TermMessage{Text: scanStderr.Text(), Source: "stderr"}
+				logDump1090TermMessage(m)
+				if shutdownES != true {
+					shutdownES = true
+				}
+			}
+			time.Sleep(1 * time.Second)
+		}
+	}
+}
+
+var dump1090ConnectionPong net.Conn = nil
+var connectionErrorPong error
+
+func pongNetworkConnection() {
+	// Send to dump1090 on port 30001
+	dump1090Addr := "127.0.0.1:30001"
```

Comment:
Not your job, but I wonder if we could, at some point, just parse the raw messages directly in Stratux, instead of our special dump1090 json format. That way, we could get rid of all this annoying dump1090 process handling for ping/pong. And potentially even the dump1090-stratux-json-format in general, allowing usage of other dump1090-like message producers like readsb etc.